log4j-api/src/main/java/org/apache/logging/log4j/status/StatusLogger.java

```diff
@@ -462,53 +718,30 @@ public boolean isEnabled(
     }
 
     @Override
-    public boolean isEnabled(final Level level, final Marker marker, final CharSequence message, final Throwable t) {
+    public boolean isEnabled(
+            final Level level, final Marker marker, final CharSequence message, final Throwable throwable) {
         return isEnabled(level, marker);
     }
 
     @Override
-    public boolean isEnabled(final Level level, final Marker marker, final Object message, final Throwable t) {
+    public boolean isEnabled(final Level level, final Marker marker, final Object message, final Throwable throwable) {
         return isEnabled(level, marker);
     }
 
     @Override
-    public boolean isEnabled(final Level level, final Marker marker, final Message message, final Throwable t) {
+    public boolean isEnabled(final Level level, final Marker marker, final Message message, final Throwable throwable) {
         return isEnabled(level, marker);
     }
 
     @Override
     public boolean isEnabled(final Level level, final Marker marker) {
-        if (DEBUG_ENABLED) {
-            return true;
-        }
-        if (listeners.size() > 0) {
-            return listenersLevel >= level.intLevel();
-        }
-        return logger.isEnabled(level, marker);
-    }
-
-    /**
-     * Queues for status events.
-     *
-     * @param <E> Object type to be stored in the queue.
-     */
-    private class BoundedQueue<E> extends ConcurrentLinkedQueue<E> {
-
-        private static final long serialVersionUID = -3945953719763255337L;
-
-        private final int size;
-
-        BoundedQueue(final int size) {
-            this.size = size;
-        }
-
-        @Override
-        public boolean add(final E object) {
-            super.add(object);
-            while (messages.size() > size) {
-                messages.poll();
-            }
-            return size > 0;
+        requireNonNull(level, "level");
+        listenerReadLock.lock();
+        try {
+            return listeners.stream()
+                    .anyMatch(listener -> listener.getStatusLevel().isLessSpecificThan(level));
```

Comment:
The maximum level of all listeners, should be cached.
Status messages are not so rare (e.g. you get one each time a new SLF4J logger is created) and the stream API will introduce a lot of garbage.

Reply:
Fixed in 21a47acf352b790fbbe6e1db6c815f05f6f8a331. Could you check that and, if it is okay, resolve the conversation, please?

Reply:
Here we might need a `public` method like `updateListenerLevels` (yes, I said `public` ;-)).

Imagine if a user updates the level of its listener, he must be able to also update the cached value (cf. Martin Fowler's [two hard things](https://martinfowler.com/bliki/TwoHardThings.html)).

Reply:
Can't they better re-register? That is, `logger.removeListener(this); logger.addListener(this)`? One less public method?

Reply:
My initial remark was mostly about `isEnabled` generating garbage (a new `Stream`).

Maybe the ideal solution is not caching as I proposed, but replacing `stream()` with a for loop:

```java
for (int i = 0; i < listeners.size(); i++) {
    if (listeners.get(i).getStatusLevel().isLessSpecificThan(level)) {
        return true;
    }
}
```

Reply:
@ppkarwasz, could you check 40cc7218ce2e1d3a56e25244ed0975fcb37ba2de, please?

Reply:
LGTM